# This page is under development
# A brief introduction to SymPortal
#### Overview
SymPortal is a analytical framework for phylogenetically resolving putative taxa within the Symbiodiniaceae using NGS data of the ITS2 amplicon. SymPortal makes use of the considerable intragenomic diversity found within every Symbiodiniaceae genome to achieve a resolution that matches that of hypervariable markers such as the chloroplastic psbAncr whilst maintaining a superior taxonomic breadth over which comparisons can be made. 

#### Resolving intra- and inter-genomic diversity: the central assumption
Because Symbiodiniaceae harbouring corals may associate with more than one taxon of Symbiodiniaceae both intragenomic and intergenomic sources of ITS2 sequence diversity may be present in coral-derived Symbiodiniaceae samples. As such, both intra- and inter-genomic sources of ITS2 seuence diversity may be present in samples. To identify set of sequences most likely derived from a single Symbiodiniaceae taxa, SymPortal works under a single biologically-derived assumption. Although corals may associate with multiple Symbiodiniaceae taxa simultaneously, within the same Symbiodiniaceae genus, corals more often than not associate with only one predominant Symbiodiniaceae taxa. As such, a set of Symbiodiniaceae ITS2 sequences found within a number of samples will become more likely to be representative of a single set taxon the more samples that set of sequences is found in. Put conversely, a set of Symbiodiniaceae ITS2 sequences found within a number of samples will become less likely to be representative of multiple taxa coincidentally harboured as similar abundances the more samples this set of sequences is found in. SymPortal therefore searches sets of Symbiodiniaceae ITS2 PCR amplicons amplified from Symbiodiniaceae-harbouring coral samples for sets of ITS2 sequences that are found across multiple samples. 

#### The ITS2 type profile: the taxonomic unit of SymPortal
Set of sequences are used to define the taxonomic unit of SymPortal, the ITS2 type profile. This ITS2 type profile is a genotype representative of a putative taxa. The sequences that are used to define these ITS2 type profiles are referred to as defining intragenomic [sequence] variants, or DIVs.

#### A community driven framework
Because SymPortal relies on finding set of sequences in multiple samples, SymPortal's ability to resolve ITS2 type profiles is dependent on the number of samples available to any given analysis. As such SymPortal's ability to resolve increases with use. It is a community driven framework. As such, each new SymPortal analysis makes use of the sequencing information associated with not only the new samples, but all previously analysed samples as well. 

#### The database
To enable this functionality the SymPortal analysis integrates with a SQL database referred to as the SymPortal database. This database is used to store the sequencing information from every previously submitted data set, and the results of analyses that have been run on these datasets. In order to cover the details of how data is submitted to the SymPortal database and how analyses are run an understanding of the structure of this relational database, and the objects (tables) that it holds is critical.

# The SymPortal relational database
For additional general information on the SymPortal database as well as instructions on how to interact with the database programmatically please refer to the wiki page [Querying the SymPortal database](https://github.com/SymPortal/SymPortal_framework/wiki/Querying-the-SymPortal-database).
## Overview
Figure 1, gives an overview of the database objects and their relations.

<p align="center">
<img src="https://github.com/didillysquat/symportal_wiki_assets/blob/master/db_schematic.png" width="70%" height="70%">
</p>

**Figure 1.** Schematic representation of the database objects (tables; bold and italicized) and the attributes used to define relations to other objects of the database (plain text associated with objects). Arrows indicate relations between objects. The objects within the database can be separated into two groups, those concerned with submission of data to the SymPortal database (data submission objects), and those associated with performing data analyses (data analysis objects).

All of the database objects have attributes. The full list can be seen in the [models.py](https://github.com/SymPortal/SymPortal_framework/blob/master/dbApp/models.py) file.

All of the database objects are related to at least one other database object. Database objects can be queried according to these relationships. 

## Data submission-based objects
The data submission-based objects are:
* _**data_set**_ - collection of samples usually from the same study. A _**data_set**_ object will generally associate to multiple _**data_set_sample**_ objects, one for every sample of the submitted data set.
* _**data_set_sample**_ - representative of a single sample. A _**data_set_sample**_ must be associated to a _**data_set**_ object and unless it contains 0 Symbiodiniaceae sequences will be associated to multiple _**data_set_sample_sequence**_ objects (one for every distinct sequence of the _**data_set_sample**_). A _**data_set_sample**_ object will also be associated with one _**clade_collection**_ object per clade represented in its collection of _**data_set_sample_sequence**_ objects. However, the summed abundance of the sequence representatives per clade must be greater than 200 for a _**clade_collection**_ object to be created.
* _**clade_collection**_ - collection of sequences of the same clade from a given sample with a summed abundance > 200. A _**clade_collection**_ must be associated to a _**data_set_sample**_ and will be associated to one _**data_set_sample_sequence**_ object for every non-distinct sequence the collection represents. A _**clade_collection**_ object may also be associated to a _**clade_collection_type**_ object if a sub-set of the sequences associated to the _**clade_collection**_ object have been evaluated to be representative of an ITS2 type profile during a SymPortal analysis. A _**clade_collection**_ object will be related to one _**clade_collection_type**_ object per ITS2 type profile identified from the sequences associated to the _**clade_collection**_ and per SymPortal analysis. As such, this object, in conjunction to the clade_collection_type object, link the data submission-based objects to the data analysis-based objects.
* _**data_set_sample_sequence**_ - a distinct ITS2 sequence found in a single sample. E.g. if the C3 sequence is returned from a sample 232 times, one data_set_sample sequence object will exist, rather than 232 separate objects. One data_set_sample_sequence object will exist for every distinct ITS2 sequence for every sample.
* _**reference_sequence**_ - an ITS2 sequence that may be found in multiple samples. The same single _**reference_sequence**_ object will represent both of two different C3 _**data_set_sample_sequence**_ objects each found in a separate sample. For example, if one sample 'A345' and a second sample 'A346' both returned the C3 sequence (abundances of 232 and 16890, respectively for each sample), the will be a _**data_set_sample_sequence**_ object for each of these sequence occurrences. However, each of these _**data_set_sample_sequence**_ objects will associate with the same single _**reference_sequence**_ object.

## Data analysis-based objects
The data analysis-based objects are:
* _**data_analysis**_ - An analysis that was run on a collection of **_data_set_** objects
* _**analysis_type**_ - An ITS2 type profile found in one or more _**clade_collection**_
* _**clade_collection_type**_ - An abstract object used to link the _**analysis_type**_ and _**clade_collection**_ objects. This object therefore represents the link between data submission-based and data analysis-based objects

# Data submission
Before any analyses are performed on samples, these samples must be converted from raw, demultiplexed illumina reads to SymPortal database objects. Once this conversion is complete, this set of objects may be used in multiple SymPortal analyses but will remain unmodified. As such the inputted data/samples can be viewed separately from the analyses that are conducted on them. This section concerns this conversion from raw data to SymPortal database objects.

### Data input format
SymPortal takes sets of demultiplexed, paired fastq.gz files as input with a forward and reverse read file for every sample. This set of submitted files will be represented in the SymPortal database as an instance of a _**data_set**_ object. 

## Sequence quality control.
Quality control (QC) of these paired files is performed on a sample by sample basis rather than working on all sequences of all samples together. Each sample contained within the data_set object is represented in the SymPortal database as a data_set_sample object. Quality control of the sequences contained in each of the samples is performed using [mothur](https://www.mothur.org/), the [blast+ suite of executables](https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=Download) and [minimum entropy decomposition](http://merenlab.org/software/med/) (MED). 

### mothur QC
The following steps are taken during the mothur-based processing:
* contiguous sequences are generated (make.contigs); 
* sequences are screened with maximum ambiguities allowed set to 0 and maximum homopolymer set to 5;
* sequences are ‘uniqued’ (distinct sequences identified); 
* singletons and doublets are removed;
* amplicons are _in silico_ PCRed using the SymVar primer pair (Hume, et al. 2018) allowing 2 forward differences and 2 reverse differences to the primers; 
* sequence direction is assessed and sequences are reverse complimented as necessary.

### blastn
#### Primary screening
The blastn programme from the blast+ suite of executables is used to identify sequence as Symbiodiniaceae or non-Symbiodiniaceae. On a sample by sample basis every sequence of the sample is run against the [symClade.fa](https://github.com/SymPortal/SymPortal_framework/blob/master/symbiodiniumDB/symClade.fa)-based blast database. 

If a sequence has an identity greater than 80% and a coverage greater than 95% to any sequence in the reference database, it is assumed to be Symbiodiniaceae in origin. 

#### Secondary screening (remote SymPortal only)
Sequences that fall below either of these thresholds are put into a bin for further taxonomic identification (this only happens on the remote instance of the SymPortal framework as access to the entire NCBI ‘nt’ database is required; as are the computational resources required to load the ‘nt’ database into memory). 

In this process, the binned sequences are run against the [NCBI’s ‘nt’ database](ftp://ftp.ncbi.nlm.nih.gov/blast/db/). The 10 closest matches are identified. If 4 or more of the matches are taxonomically annotated as either Symbiodinium or Symbiodiniaceae, the percentage coverage is above 95%, and the percentage identity match is above 60%, then this sequence is considered Symbiodiniaceae in origin (sequences must also be between 184 and 310bp in length; more below).

Importantly, all sequences identified as Symbiodiniaceae in origin at this stage are added to [the symClade.fa fasta](https://github.com/SymPortal/SymPortal_framework/blob/master/symbiodiniumDB/symClade.fa) and the related database objects are remade. The screening of the sequences is then repeated iteratively until no new Symbiodiniaceae sequences are found. Sequences deemed to be Symbiodiniaceae in origin are carried forwards in the analysis. Those sequences deemed to be non-Symbiodiniaceae in origin are written to disk (in the directory of the paired fastq.gz files) for access by the user.

### Size screening
After taxonomic screening Symbiodiniaceae sequences are size screened. This is done with hard cut-offs that were determined empirically as 50 bp +- the average size of the smallest and largest clades (clade A 234 bp, cladeB 266). These cutoffs are max=310, min=184. At this stage, the Symbiodiniaceae sequences are written to disk separated by taxonomic clade (A-I) in preparation for MED analysis.

### MED decomposition
A [MED decomposition is run](https://www.nature.com/articles/ismej2014195) using a dynamic M value set as the largest of 4 or 0.004 * the number of sequences being decomposed. MED nodes are read in from the analysis output and are eventually used to create the _**data_set_sample_sequence**_ and _**reference_sequence**_ objects of the database. However, before we take a closer look at these two objects we must first take a closer look at the _**clade_collection**_ object.

### An introduction to the _**clade_collection**_, _**data_set_sample_sequence**_ and _**reference_sequence**_ objects
Broadly speaking, in the field of Symbiodiniaceae taxonomy (specifically Symbiodiniaceae phylogenetics) all Symbiodiniaceae taxa defined using the ITS2 marker are resolved at, or below, the clade level. Therefore, no Symbiodiniaceae taxa defined with the ITS2 markers may contain intragenomic variants from multiple clades (although multiple Symbiodiniaceae taxa from different clades may reside in a single host). As such, SymPortal subdivides the ITS2 amplicon sequences found in every sample into clade groupings. SymPortal will only attempt to discover ITS2 type profiles (the taxonomic unit of resolution outputted from SymPortal analyses) from any such sample's clade grouping if it contains more than 200 sequences. Any cut-off lower than this may lead to an inability to detect defining intragenomic variants due to insufficient sequencing depth rather than true biological absence . Each collection of these clade grouped sequences from a given sample is represented in the SymPortal database as a _**clade_collection**_. Given that an analysis will not attempt to search a clade group where the total abundance of the constituent sequences is not above 200, per sample, _**clade_collection**_ objects will only be made for clade-separated groups of sequences above this threshold. E.g. if a sample contains 1980 clade C sequences, 42192 clade D sequences and 124 clade A sequences, it will have a _**clade_collection**_ object for each of the clade C and clade D sequences but there will not be a _**clade_collection**_ object representing the clade A sequences. The sample, will contain two clade collections, despite harbouring representative sequences from three of the Symbiodiniaceae clades.

An instance of a _**clade_collection**_ will contain a number of ITS2 amplicon sequences. To reduce information redundancy, multiple occurrences of the same sequence associated with a _**clade_collection**_ are stored in a single object. For example, 100 C3 sequences found in a _**clade_collection**_ will be represented as a single sequence instance found 100 times rather than 100 sequence instances. As well as being found multiple times within the same sample, sequences will be found in common between many different samples. For example, the previously mentioned C3 sequence is found globally. To minimise information redundancy, information specific to an instance of a sequence, e.g. which _**clade_collection**_ it was found in and at what abundance it was found at, is stored separately from the sequence information e.g. the nucleotide sequence, the clade, and the sequence name. The _**clade_collection**_-specific object is the _**data_set_sample_sequence**_ whilst the general sequence information is the _**reference_sequence**_ object. This concept is illustrated in figure 2.

<p align="center">
<img src="https://github.com/didillysquat/symportal_wiki_assets/blob/master/sequenceNaming_noCC.png" width="50%" height="50%">
</p>

**Figure 2.** Schematic representation of the relationship between the _**data_set_sample**_, _**clade_collection**_, _**data_set_sample_sequence**_ and _**reference_sequence**_ objects within the context of minimising redundancy when storing sequence information.

### Naming of ITS2 sequences
Due to the massive diversity of ITS2 sequences that have already been submitted to the SymPortal database, not all sequences are given names. Only those sequences that are used in the definition of ITS2 type profiles (i.e. DIVs) are named. This naming process currently only happens in the remote version SymPortal to prevent disagreement with novel sequences named by local instance. Given that only those sequences that are used to define ITS2 type profiles are named, the naming process doesn't happen during data submission but rather at the end of a data analysis. This is due to the fact that it is impossible to know whether any novel sequences encountered (those sequences not already in the SymPortal database) during data submission will be used to define ITS2 type profiles before a data analysis has been run. However, during data submission, each of the MED nodes output from the MED analysis are checked against the current reference_sequence objects held in the SymPortal database. If the MED node sequences already match one of sequences of an existing reference_sequence, the data_set_sample_sequence generated from this MED node will be associated to this reference_sequence. Importantly, when looking for matches to the output MED nodes, subsetting is taken into account. For example, if a MED node sequence is AGGATGCA and there is a _**reference_sequence**_ object with a `**reference.sequence**` sequence of GGATGCA then these will be considered a match, and the _**data_set_sample_sequence**_ generated will be associated to the _**reference_sequence**_ with the GGATGCA sequence. If there is no match for a given MED node sequence, a new _**reference_sequence**_ object is created. This _**reference_sequence**_ will not have a name. Unnamed _**reference_sequence**_ objects are referred to by their unique identifier (UID).

### Data submission conclusion and output
Once MED nodes have been used to generate the _**data_set_sample_sequence**_ objects, that are in turn associated to the _**reference_sequence**_, _**clade_collection**_, and _**data_set_sample**_ objects, that are all associated to a single _**data_set**_ object, data submission is complete.

# Data analysis
## Overview
A SymPortal data analysis is the process in which ITS2 type profiles are predicted for a selection of _**data_set**_ objects and their associated samples and sequences. A SymPortal data analysis can be divided into two phases: ITS2 type profile discovery and ITS2 type profile assignment. 

Briefly, in the discovery phase, sets of sequences found in samples are searched for in all other samples to identify sets of sequences as large as possible that are found in common between as many samples as possible. Sets of sequences that are shared between a number of samples greater than a set minimum threshold are used to define ITS2 type profiles. That is, for each set of sequences, each sequence becomes a DIV for the ITS2 type profile being defined. Also the maximum and minimum relative abundances of each DIV (across all of the samples in which the set of sequences in question was found) is assigned to the ITS2 type profile and used for identifying the ITS2 type profile in the ITS2 type profile assignment phase.

In the assignment phase, every _**clade_collection**_ (every sample) is searched to see whether the DIVs of the ITS2 type profiles identified in the discovery phase can be found, and if so, whether the relative abundances at which these DIVs are found are within the defining relative abundances of the ITS2 type profile in question. A more detailed description of the discovery and the assignment phases are given below.

## ITS2 type profile discovery
The purpose of ITS2 type profile discovery is to find a set of sequences from the submitted _**data_set_sample**_-associated _**clade_collection**_ objects where each set of sequences is as large as possible whilst still maintaining the minimum level of support (number of _**clade_collection**_ objects the set of sequences is found in). Support is quantified as the number of _**clade_collection**_ objects in which the set of sequences is found. Currently a set of sequences must be found in at least 4 `**clade_collection**` objects to be considered supported and therefore go on to be used to define an ITS2 type profile. Each _**clade_collection**_ can only count towards the support of one ITS2 type profile. An easy way to think about this is that a putative ITS2 type profile is assigned to a _**clade_collection**_ in type discovery.

### Creation of initial profiles for testing
For each _**clade_collection**_ in the analysis, an initial profile is identified, which will then be tested to see if it is supported, i.e. found in other _**clade_collection**_ objects. Not all sequences in a _**clade_collection**_ are used to create this initial profile rather, only sequences found at a relative abundance above 3% are considered (relative to the total number of sequences in the _**clade_collection**_). This 3% cutoff is referred to as the withinCladeCutoff. The 3% value of the withinCladeCutoff has been empirically determined. Higher values begin to limit resolving power. Lower values quickly increase the number of sequences included in the initial type profiles. This increases computational effort considerably, reduces the probability of finding profiles in common between samples, and increases the probability of non-discovery of DIVs due to sequencing depth artefacts rather than true biological absence.

### Searching for supported type profiles
The above process generates a list of initial type profiles, one for each _**clade_collection**_. For each putative type profile the number of _**clade_collection**_ it was found in and the most abundant sequence of each _**clade_collection**_ it was found in (referred to as the majority sequence or majSeq) are kept track of. Two initial profiles are considered identical if they contain the same set of DIVs, i.e. the abundance of DIVs is not taken into account at this point.

Searching for support for the above list of initial profiles is done clade by clade, i.e. supported types from clade A are determined first, then clade B etc. You might imagine that determining support for the list of profiles above is simple and would follow the following logic: for every putative type profile found, see how many _**clade_collection**_ it was found in, if enough, consider this type profile supported. However, a large number of initial profiles will not gain support from this method alone, let’s call these profiles ‘unsupported’. SymPortal works with these ‘unsupported’ profiles by evaluating permutations of shorter subsets of the profiles’ sequences (essentially collapsing them) until additional support is found. See section [A.1 Determining supported profiles]() for further detail.

Once a list of supported profiles has been generated, these profiles are represented by _**analysis_type**_ objects in the database.

Each _**analysis_type**_ has a name that is made up of the defining intragenomic variants. How each of these DIVs is named will be covered later, but the format of the _**analysis_type**_ name will be explained here. The DIVs in a _**analysis_type**_ name are listed in order of total abundance across all the _**clade_collection**_ objects the _**analysis_type**_ was found in. For example, a typical _**analysis_type**_ name might look like, C3-C3a-C3cc, where C3 is the most abundant sequence found and C3cc the least. In this example, C3 was the most abundant sequence (majSeq) of the three DIVs in each of the _**clade_collection**_ objects the _**analysis_type**_ was found in. In some cases, the most abundant DIV of a type might vary from _**clade_collection**_ to _**clade_collection**_. For example, let’s examine the _**analysis_type**_ C3/C3c-C3cc. This _**analysis_type**_ was found in 10 _**clade_collection**_ objects. In 7 of the _**clade_collection**_ objects, C3 was the most abundant sequence (majSeq) of the three DIVs. However, in 3 of the _**clade_collection**_ objects, C3c was the most abundant. This so-called co-dominance is denoted by the ‘/’. Co-dominant DIVs are always listed in the order of the number of _**clade_collection**_ objects they are the majSeq in. After the co-dominant sequences have been listed in a name, the other DIVs are listed in order of total abundance as described in the first naming example.

